server/manager/src/test/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueueTest.java

```diff
@@ -0,0 +1,185 @@
+    EasyMock.expect(cj2.getFiles()).andReturn(Set.of(file2, file3, file4)).anyTimes();
+
+    EasyMock.replay(tm, cj1, cj2);
+
+    CompactionJobPriorityQueue queue = new CompactionJobPriorityQueue(GROUP, 2);
+    assertTrue(queue.add(tm, List.of(cj1, cj2)));
+
+    EasyMock.verify(tm, cj1, cj2);
+
+    assertEquals(5L, queue.getLowestPriority());
+    assertEquals(2, queue.getMaxSize());
+    assertEquals(0, queue.getDequeuedJobs());
+    assertEquals(0, queue.getRejectedJobs());
+    assertEquals(2, queue.getQueuedJobs());
+    MetaJob job = queue.poll();
+    assertEquals(cj1, job.getJob());
+    assertEquals(tm, job.getTabletMetadata());
+    assertEquals(1, queue.getDequeuedJobs());
+
+  }
+
+  @Test
+  public void testAddMoreThanMax() {
+
+    CompactableFile file1 = EasyMock.createMock(CompactableFileImpl.class);
+    CompactableFile file2 = EasyMock.createMock(CompactableFileImpl.class);
+    CompactableFile file3 = EasyMock.createMock(CompactableFileImpl.class);
+    CompactableFile file4 = EasyMock.createMock(CompactableFileImpl.class);
+    CompactableFile file5 = EasyMock.createMock(CompactableFileImpl.class);
+    CompactableFile file6 = EasyMock.createMock(CompactableFileImpl.class);
+
+    KeyExtent extent = new KeyExtent(TableId.of("1"), new Text("z"), new Text("a"));
+    TabletMetadata tm = EasyMock.createMock(TabletMetadata.class);
+    EasyMock.expect(tm.getExtent()).andReturn(extent).anyTimes();
+
+    CompactionJob cj1 = EasyMock.createMock(CompactionJob.class);
+    EasyMock.expect(cj1.getGroup()).andReturn(GROUP).anyTimes();
+    EasyMock.expect(cj1.getPriority()).andReturn((short) 10).anyTimes();
+    EasyMock.expect(cj1.getFiles()).andReturn(Set.of(file1)).anyTimes();
+
+    CompactionJob cj2 = EasyMock.createMock(CompactionJob.class);
+    EasyMock.expect(cj2.getGroup()).andReturn(GROUP).anyTimes();
+    EasyMock.expect(cj2.getPriority()).andReturn((short) 5).anyTimes();
+    EasyMock.expect(cj2.getFiles()).andReturn(Set.of(file2, file3, file4)).anyTimes();
+
+    CompactionJob cj3 = EasyMock.createMock(CompactionJob.class);
+    EasyMock.expect(cj3.getGroup()).andReturn(GROUP).anyTimes();
+    EasyMock.expect(cj3.getPriority()).andReturn((short) 1).anyTimes();
+    EasyMock.expect(cj3.getFiles()).andReturn(Set.of(file5, file6)).anyTimes();
+
+    EasyMock.replay(tm, cj1, cj2, cj3);
+
+    CompactionJobPriorityQueue queue = new CompactionJobPriorityQueue(GROUP, 2);
+    assertTrue(queue.add(tm, List.of(cj1, cj2, cj3)));
+
+    EasyMock.verify(tm, cj1, cj2, cj3);
+
+    assertEquals(5L, queue.getLowestPriority());
+    assertEquals(2, queue.getMaxSize());
+    assertEquals(0, queue.getDequeuedJobs());
+    assertEquals(1, queue.getRejectedJobs());
+    assertEquals(2, queue.getQueuedJobs());
+    MetaJob job = queue.poll();
+    assertEquals(cj1, job.getJob());
+    assertEquals(tm, job.getTabletMetadata());
+    assertEquals(1, queue.getDequeuedJobs());
+
```

Comment:
Would be good to poll till empty here too.

Reply:
Implemented in b68ab6a